PWGHF/D2H/Tasks/taskXic.cxx

```diff
@@ -31,175 +31,277 @@ using namespace o2::analysis::hf_cuts_xic_to_p_k_pi;
+      registry.fill(HIST("Data/hPVsTPCNSigmaPi_Prong0"), prong0_P, trackProng0.tpcNSigmaPi());
+      registry.fill(HIST("Data/hPVsTPCNSigmaKa_Prong0"), prong0_P, trackProng0.tpcNSigmaKa());
+
+      registry.fill(HIST("Data/hPVsTPCNSigmaPr_Prong1"), prong1_P, trackProng1.tpcNSigmaPr());
+      registry.fill(HIST("Data/hPVsTPCNSigmaPi_Prong1"), prong1_P, trackProng1.tpcNSigmaPi());
+      registry.fill(HIST("Data/hPVsTPCNSigmaKa_Prong1"), prong1_P, trackProng1.tpcNSigmaKa());
+
+      registry.fill(HIST("Data/hPVsTPCNSigmaPr_Prong2"), prong2_P, trackProng2.tpcNSigmaPr());
+      registry.fill(HIST("Data/hPVsTPCNSigmaPi_Prong2"), prong2_P, trackProng2.tpcNSigmaPi());
+      registry.fill(HIST("Data/hPVsTPCNSigmaKa_Prong2"), prong2_P, trackProng2.tpcNSigmaKa());
+
+      // TOF nSigma histograms
+      registry.fill(HIST("Data/hPVsTOFNSigmaPr_Prong0"), prong0_P, trackProng0.tofNSigmaPr());
+      registry.fill(HIST("Data/hPVsTOFNSigmaPi_Prong0"), prong0_P, trackProng0.tofNSigmaPi());
+      registry.fill(HIST("Data/hPVsTOFNSigmaKa_Prong0"), prong0_P, trackProng0.tofNSigmaKa());
+
+      registry.fill(HIST("Data/hPVsTOFNSigmaPr_Prong1"), prong1_P, trackProng1.tofNSigmaPr());
+      registry.fill(HIST("Data/hPVsTOFNSigmaPi_Prong1"), prong1_P, trackProng1.tofNSigmaPi());
+      registry.fill(HIST("Data/hPVsTOFNSigmaKa_Prong1"), prong1_P, trackProng1.tofNSigmaKa());
+
+      registry.fill(HIST("Data/hPVsTOFNSigmaPr_Prong2"), prong2_P, trackProng2.tofNSigmaPr());
+      registry.fill(HIST("Data/hPVsTOFNSigmaPi_Prong2"), prong2_P, trackProng2.tofNSigmaPi());
+      registry.fill(HIST("Data/hPVsTOFNSigmaKa_Prong2"), prong2_P, trackProng2.tofNSigmaKa());
     }
   }
-
-  void processMc(soa::Join<aod::HfCand3Prong, aod::HfSelXicToPKPi, aod::HfCand3ProngMcRec> const& candidates,
-                 soa::Join<aod::McParticles, aod::HfCand3ProngMcGen> const& particlesMC, aod::BigTracksMC const&)
+  // Fill MC histograms
+  void processMc(soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelXicToPKPi, aod::HfCand3ProngMcRec>> const& candidates,
+                 soa::Join<aod::McParticles, aod::HfCand3ProngMcGen> const& particlesMC, aod::BigTracksMC const& /*tracks*/)
   {
+
     // MC rec.
-    for (auto& candidate : selectedMCXicCandidates) {
+    for (auto& candidate : candidates) {
+      // Selected Xic
       if (!(candidate.hfflag() & 1 << DecayType::XicToPKPi)) {
         continue;
-      }
+      } // rapidity selection
       if (yCandMax >= 0. && std::abs(yXic(candidate)) > yCandMax) {
         continue;
       }
 
       auto mass = 0.;
       auto massC = 0.;
+      auto candidatePt = candidate.pt();
 
       if (candidate.isSelXicToPKPi() >= selectionFlagXic) {
         mass = invMassXicToPKPi(candidate);
       }
       if (candidate.isSelXicToPiKP() >= selectionFlagXic) {
-        massC = invMassXicToPiKP(candidate);
+        massC = invMassXicToPiKP(candidate); // mass conjiugated
```

Comment:
The names `mass` and `massC` are not clear enough. Please rename to something better.